media-libs/libheif/libheif-1.18.0.ebuild

```diff
@@ -0,0 +1,104 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake xdg multilib-minimal
+
+if [[ ${PV} == *9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/strukturag/libheif.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/strukturag/libheif/releases/download/v${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
+fi
+
+DESCRIPTION="ISO/IEC 23008-12:2017 HEIF file format decoder and encoder"
+HOMEPAGE="https://github.com/strukturag/libheif"
+
+LICENSE="GPL-3"
+SLOT="0/$(ver_cut 1-2)"
+IUSE="+aom gdk-pixbuf go rav1e svt-av1 test +threads x265"
+REQUIRED_USE="test? ( go )"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	test? (
+		dev-cpp/catch
+		dev-lang/go
+	)
+"
+DEPEND="
+	media-libs/dav1d:=[${MULTILIB_USEDEP}]
+	media-libs/libde265:=[${MULTILIB_USEDEP}]
+	media-libs/libpng:0=[${MULTILIB_USEDEP}]
+	media-libs/tiff:=[${MULTILIB_USEDEP}]
+	sys-libs/zlib:=[${MULTILIB_USEDEP}]
+	media-libs/libjpeg-turbo:0=[${MULTILIB_USEDEP}]
+	aom? ( >=media-libs/libaom-2.0.0:=[${MULTILIB_USEDEP}] )
+	gdk-pixbuf? ( x11-libs/gdk-pixbuf[${MULTILIB_USEDEP}] )
+	go? ( dev-lang/go )
+	rav1e? ( media-video/rav1e:= )
+	svt-av1? ( media-libs/svt-av1[${MULTILIB_USEDEP}] )
+	x265? ( media-libs/x265:=[${MULTILIB_USEDEP}] )"
+RDEPEND="${DEPEND}"
+
+MULTILIB_WRAPPED_HEADERS=(
+	/usr/include/libheif/heif_version.h
+)
+
+src_prepare() {
+	default
+
+	# https://github.com/strukturag/libheif/issues/1249
+	sed -i 's/ ${CMAKE_INSTALL_PREFIX}/ ${DESTDIR}${CMAKE_INSTALL_PREFIX}/g' "${WORKDIR}"/"${P}"/examples/CMakeLists.txt
+
+	if use test ; then
+		# bug 865351
+		rm tests/catch.hpp || die
+		ln -s "${ESYSROOT}"/usr/include/catch2/catch.hpp tests/catch.hpp || die
+	fi
+
+	cmake_src_prepare
+
+	multilib_copy_sources
+}
+
+multilib_src_configure() {
+	export GO111MODULE=auto
+	local mycmakeargs=(
+		# https://github.com/strukturag/libheif/issues/1249
+		-DDESTDIR="${D}"
```

Comment:
Is DESTDIR not being set otherwise? I am a bit surprised that it would be the only cmake package needing to explicitly set DDESTDIR in the tree 🤔

Reply:
Is the solution implemented by upstream, the only way I know to define cmake vars is to pass thru -D

Reply:
No, the upstream solution is faulty. It must not rely on DESTDIR as a cmake variable defined in src_configure -- it needs to look up the environment variable as part of the cmake code executed at src_install time.

Reply:
Ok, waiting for final upstream solution to deliver ebuild

Reply:
Added consolidated upstream patches in https://github.com/gentoo/gentoo/pull/36951/commits/7433cdaa5a8383c98c4f3f95b26444a46c371f72